lib/maze/accessors/trace_accessors.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Maze
+  module Accessors
+    # Provides functions for accessing and testing traces and spans 
+    class TraceAccessors
+      class << self
+        # Traces
+
+        # Retrieves a value within a trace
+        #
+        # @param list [String] The name of the list to get spans from
+        # @param field [String] The field to check the attributes of
+        # @param attribute [String] The attribute to check
+        #
+        # @returns [String|Integer] A string or integer representation of the attribute
+        def attribute_by_key(list, field, attribute)
```

Comment:
Just to demonstrate the idea of what we discussed IRL earlier (which I think you fully grasped anyway), this would be a good example for a method to sit on a `Trace` class.